### PR TITLE
Begin recommendation system

### PR DESCRIPTION
**Pull request by RealHinome**
Hello,

This PR is just to begin the recommendation system. AI need to be done later.

### CODE REVIEW

File: recommendation/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<img src="https://avatars.githubusercontent.com/u/81774317?s=200&v=4" width="40" />
```
Comment (ncs-pl) [inline recommendation/README.md:1]: Why using HTML instead of `Alt`

File: recommendation/model/Post.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package model
+
```
Comment (ncs-pl) [inline recommendation/model/Post.go:2]: Add documentation

File: recommendation/router/index.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package router
+
+import (
+	"fmt"
+	"net/http"
+)
+
```
Comment (ncs-pl) [inline recommendation/router/index.go:7]: Add documentation

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -22,9 +22,9 @@ jobs:
         go-version: 1.19
```
Comment (ncs-pl) [inline .github/workflows/go.yml]: Your `go.mod` file declares

```gomod 
go 1.20
```

yet you are using Go 1.19 in CI, is that normal?

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -22,9 +22,9 @@ jobs:
...
 
     - name: Build
-      working-directory: ./rest
+      working-directory: ${{ matrix.dir }}
       run: go build -v ./...
 
     - name: Test
-      working-directory: ./rest
+      working-directory: ${{ matrix.dir }}
       run: go test -v ./...
```
Comment (ncs-pl) [inline .github/workflows/go.yml:30]: You should add lint checking step using `gofmt`

File: recommendation/helpers/jwt.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+func Check(token string) (string, error) {
+	var key string
+	if os.Getenv("JWT_SECRET") != "" {
+		key = os.Getenv("JWT_SECRET")
+	} else {
+		key = "secret"
+	}
+
+	hsverifier, err := jwt.NewVerifierHS(jwt.HS512, []byte(key))
+	if err != nil {
```
Comment (ncs-pl) [inline recommendation/helpers/jwt.go:22]: It is more idiomatic to write 

```go
if err = foo(); err != nil {
  // ...
}
```

File: recommendation/main.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/joho/godotenv"
+	"github.com/robfig/cron"
+)
+
+func main() {
+	// Get key-value in .env file
+	godotenv.Load()
+
+	// Start a new cron job
+	c := cron.New()
```
Comment (ncs-pl) [inline recommendation/main.go:21]: In the future, you should use separate Cron jobs, such as Kubernetes' one, as they have specific properties and configuration settings that are preferable

File: recommendation/router/recommendation.go [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/Gravitalia/recommendation/database"
+	"github.com/Gravitalia/recommendation/helpers"
+	"github.com/Gravitalia/recommendation/model"
+)
+
```
Comment (ncs-pl) [inline recommendation/router/recommendation.go:11]: Add Documentation

File: recommendation/router/recommendation.go [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+func Get(w http.ResponseWriter, req *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	jsonEncoder := json.NewEncoder(w)
+
+	var vanity string
+	if req.Header.Get("authorization") == "" {
+		w.WriteHeader(http.StatusUnauthorized)
```
Comment (ncs-pl) [inline recommendation/router/recommendation.go:20]: We usually prefer the status `Unauthenticated` in situations where auth info are not specified

**Review by ncs-pl — CHANGES_REQUESTED** [review]
Nice work, I requested some changes.

You should also work on logging and tracing.
Indeed, the code is clearly missing log information (such as HTTP requests etc.)

You should also have a look at Tracing components (Prometheus, OpenTelemetry...) and implement them.

File: recommendation/README.md (hunk 1 above)
Comment (RealHinome) [reply recommendation/README.md:1]: I need to set a width (else the image is too big)

File: .github/workflows/go.yml (hunk 4 above)
Comment (RealHinome) [reply .github/workflows/go.yml]: An error between the local and the CI. I'm changing the CI right away!

File: recommendation/router/recommendation.go (hunk 9 above)
Comment (RealHinome) [reply recommendation/router/recommendation.go:20]: It seems that there is no HTTP code for Unauthenticated error. May have you done a typing error?

**Comment by RealHinome** [comment]
As soon as I have time (probably next week), I'll look into tracing components. Not sure if it will be implemented for several months though.

**Review by ncs-pl — APPROVED** [review]

**Pull request merged**